worlds/oot_soh/location_access/dungeons/ganons_castle.py

```diff
@@ -5,22 +5,350 @@
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT1, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT2, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT3, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT4, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT5, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT6, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT7, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT8, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT9, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT10, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT11, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT12, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT13, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT14, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT15, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT16, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT17, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT18, lambda bundle: can_break_pots(bundle)),
+    ])
+    # Connections
+    connect_regions(Regions.GANONS_TOWER_BEFORE_GANONDORFS_LAIR, world, [
+        (Regions.GANONS_TOWER_FLOOR_3, lambda bundle: True),
+        (Regions.GANONDORFS_LAIR, lambda bundle: has_item(Items.GANONS_CASTLE_BOSS_KEY, bundle))
+    ])
+    
+    ## Ganondorf's Lair
+    # Locations
+    add_locations(Regions.GANONDORFS_LAIR, world, [
+        (Locations.GANONS_CASTLE_GANONDORF_HINT, lambda bundle: has_boss_soul(Items.GANONS_SOUL, bundle))
+    ])
+    # Connections
+    connect_regions(Regions.GANONDORFS_LAIR, world, [
+        (Regions.GANONS_CASTLE_ESCAPE, lambda bundle: can_kill_enemy(bundle, Enemies.GANONDORF))
+    ])
+    
+    ## Ganon's Castle Escape
+    # Locations
+    #add_locations(Regions.GANONS_CASTLE_ESCAPE, world, [
+    #   # 10 pots
+    #])
+    # Connections
+    connect_regions(Regions.GANONS_CASTLE_ESCAPE, world, [
+        (Regions.GANONS_ARENA, lambda bundle: True)
+    ])
+    
+    ## Ganon's Arena
+    # Events
     add_events(Regions.GANONS_ARENA, world, [
         (EventLocations.GANON_DEFEATED, Events.GAME_COMPLETED, lambda bundle: 
          (can_use(Items.LIGHT_ARROW, bundle) and can_use(Items.MASTER_SWORD, bundle)
           and not world.options.triforce_hunt)
          or has_item(Events.GAME_COMPLETED, bundle))
     ])
```

Comment:
You'll want to replace `(can_use(Items.LIGHT_ARROW, bundle) and can_use(Items.MASTER_SWORD, bundle)` with `can_kill_enemy(bundle, Enemies.GANON)`. I just added this before to test generation, so the final logic is slightly different. The rest of the rule does hold up though.